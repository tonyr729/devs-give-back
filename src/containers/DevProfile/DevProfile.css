.profile-background {
  background-color: #ffffff;
  width: 100vw;
  height: 100vh; }

.dev-header {
  height: 9vh;
  width: 100vw;
  padding-top: 4rem;
  background-color: #03a9f4; }

.dev-header-container {
  height: 100%;
  width: 80rem;
  display: block;
  margin: 0 auto;
  background-color: #03a9f4; }
  .dev-header-container img {
    border-radius: 3rem;
    width: 9rem;
    margin: 3rem 2rem 0; }
  .dev-header-container p {
    font-size: 2rem;
    text-align: -webkit-auto;
    font-weight: 100;
    top: 9rem; }
  .dev-header-container ul {
    list-style: none; }
  .dev-header-container li {
    display: inline;
    margin-right: 1rem;
    font-size: .8rem;
    color: #f8f8ff;
    position: relative;
    top: 7.5rem;
    left: 9.9rem; }

.main {
  height: 40vh;
  width: 100vw;
  background-color: #ffffff; }

.main-container {
  height: 80vh;
  width: 80rem;
  display: block;
  margin: 0 auto;
  background-color: #fff; }

.main-section-titles {
  font-size: 3rem;
  padding-top: 2rem;
  font-weight: 100;
  width: 60rem; }
  .main-section-titles a {
    color: #ffffff;
    padding: 0 4rem; }
    .main-section-titles a:hover {
      color: #00364e; }
    .main-section-titles a:active {
      border-bottom: 1px solid #ffffff; }

.info-text {
  background-color: #fff;
  width: 100%;
  height: 100%; }
  .info-text p {
    font-size: 3rem;
    padding-top: 3rem;
    line-height: 3rem;
    color: #de5858; }

.current-projects {
  background-color: #fff;
  width: 59rem;
  height: 34rem;
  margin: 4rem auto 0;
  overflow: scroll; }
  .current-projects button {
    border: 1px solid;
    font-size: .8rem;
    padding: .2rem .8rem; }
    .current-projects button:hover {
      background-color: #1db6fc; }

.project-card {
  background-color: #d2effd;
  width: 59rem;
  padding-top: .8rem;
  height: 5rem;
  margin-top: 1rem; }

.dev-project-title {
  color: #737373;
  text-align: left;
  font-size: 1.5rem;
  font-weight: 100; }

.project-button {
  margin-left: 1.5rem; }
