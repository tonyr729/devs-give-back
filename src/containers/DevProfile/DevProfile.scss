.profile-background {
  background-color: #1db6fc;
  width: 100vw;
  height: 100vh;
}

.dev-header {
  height: 20vh;
  width: 100vw;
  background-color: #1db6fc;
}

.dev-header-container {
  height:100%;
  width:80rem;
  display: block;
  margin: 0 auto;
  background-color: #03a9f4;
  img {
    border-radius: 3rem;
    width: 9rem;
    margin: 3rem 2rem 0;
  };
  p {
    font-size: 2rem; 
    text-align: -webkit-auto; 
    position: relative; 
    font-weight: 100;
    top: 9rem; 
  };
  ul {
    list-style: none;
  };
  li {
    display: inline;
    margin-right: 1rem;
    font-size: .8rem;
    color: #f8f8ff;
    position: relative;
    top: 7.5rem;
    left: 9.9rem;
  };
}

.main {
  height: 40vh;
  width: 100vw;
  background-color: #1db6fc;
}

.main-container {
  height:80vh;
  width:80rem;
  display: block;
  margin: 0 auto;
  background-color: #fff;
}

.main-section-titles {
  font-size: 3rem;
  padding-top: 2rem;
  font-weight: 100;
  margin-left: 14rem;
  width: 60rem;
  color: #000;
  a {
    color: #7c7c7c;
    &:hover {
      color: #03a9f4;
    }
  }   
}

.info-text {
  background-color: #fff;
  width: 100%;
  height: 100%;
  p {
    font-size: 3rem;
    padding-top: 3rem;
    line-height: 3rem;
    color: #de5858;
  }
}

.current-projects {
  background-color: #fff;
  width: 59rem;
  height: 34rem;
  margin: 4rem auto 0 14.5rem;
  overflow: scroll;
  button {
    border: 1px solid;
    font-size: .8rem;
    padding: .2rem .8rem;
    &:hover {
      background-color: #1db6fc;
    }
  }
}

.project-card {
  background-color: #d2effd;
  width: 59rem;
  padding-top: .8rem;
  height: 5rem;
  margin-top: 1rem;
}

.dev-project-title {
  color: #737373;
  text-align: left;
  font-size: 1.5rem;
  font-weight: 100;
}

.project-button {
  margin-left: 1.5rem;
}

.signout-button {
  background: none;
  display: block;
  margin: .5rem 0 0 4.2rem;
  font-size: 1rem;
  color: #07a8f4;
  &:hover {
    color: #800080
  }
}