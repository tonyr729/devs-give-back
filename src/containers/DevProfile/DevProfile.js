import React, { Component } from 'react';
import './DevProfile.css';
import { connect } from 'react-redux';
import firebase from '../../firebase/firebase';
import { findMatchingProblem } from '../../helpers/apiCalls';


class ClientProfile extends Component {

  runFetch = async () => {
    const url = `https://api.github.com/repos/tonyr729/movie-tracker/stats/contributors?access_token=${this.props.dev.token}`;
    const response = await fetch(url);
    const data = response.json();
    console.log(data)
  }

  logInCheck = (dev) => {
    const value = Object.keys(dev).length;
    if ( value === 0 ) {
      this.props.history.push('/dev-login')
    }
  }

  render() {
    this.logInCheck(this.props.dev)
    return (
      <div>
        <p>This is the DevProfile</p>
        <button onClick={this.runFetch}>Run Fetch</button>
      </div>
    )
  }
}

const mapStateToProps = (state) => ({
  dev: state.dev
});


export default connect(mapStateToProps)(ClientProfile);