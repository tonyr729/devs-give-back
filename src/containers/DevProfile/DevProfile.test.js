import { DevProfile, mapStateToProps, mapDispatchToProps} from './DevProfile';
import { shallow } from 'enzyme';
import React from 'react';
import { Redirect } from 'react-router-dom';


describe('DevProfile', () => {
  it('should match snapshot', () => {
    const mockProps = {
      dev: {
        name: 'Tony'
      },
      addProjects: jest.fn()
    }
    const wrapper = shallow(<DevProfile {...mockProps} />);

    expect(wrapper).toMatchSnapshot();
  });

  describe('displayProjects', () => {
    let mockProps;
    let wrapper;

    beforeEach(() => {
      mockProps = {
        dev: {
          name: 'Tony'
        },
        projects: [{title: 'title'}, {title: 'title'}],
        addProjects: jest.fn()
      };
      wrapper = shallow(<DevProfile {...mockProps} />);
    })

    it('should return project cards if projects exists', () => {
      expect(wrapper.find('.project-card').length).toEqual(2)
    });

    it('should return message if no projects exist', () => {
      const mockProps = {
        dev: {
          name: 'Tony'
        },
        projects: [],
        addProjects: jest.fn()
      };
      const wrapper = shallow(<DevProfile {...mockProps} />);
      expect(wrapper.find('.info-text').length).toEqual(1)
    });
  });

  describe('statusCheck', () => {
    let mockProps;
    let wrapper;

    beforeEach(() => {
      mockProps = {
        dev: {
          name: 'Tony'
        },
        projects: [{title: 'title'}, {title: 'title'}],
        addProjects: jest.fn()
      };
      wrapper = shallow(<DevProfile {...mockProps} />);
    })

    it('should redirect if there is no dev', () => {
      const expected = (<Redirect to='/dev-login'/>)
      const actual = wrapper.instance().statusCheck({})
      expect(actual).toEqual(expected)
    });
  });

  describe('mapDispatchToProps', () => {
    it('should call dispatch on signInDev with the correct params', () => {

      const mockDispatch = jest.fn();
      const user = {name: 'Tony'}
      const mappedProps = mapDispatchToProps(mockDispatch);
      const mockAction = {
        type: 'SIGN_IN_DEV',
        user
      };
      
      mappedProps.signInDev(user);
      expect(mockDispatch).toHaveBeenCalledWith(mockAction);
      
    });

    it('should call dispatch on addProjects with the correct params', () => {

      const mockDispatch = jest.fn();
      const projects = {projects: 'projects'}
      const mappedProps = mapDispatchToProps(mockDispatch);
      const mockAction = {
        type: 'ADD_PROJECTS',
        projects
      };
      
      mappedProps.addProjects(projects);
      expect(mockDispatch).toHaveBeenCalledWith(mockAction);
      
    });
  });
});