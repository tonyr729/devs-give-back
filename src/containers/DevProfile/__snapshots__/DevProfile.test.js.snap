// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DevProfile should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DevProfile
    addProjects={[Function]}
    dev={
        Object {
            "name": "Tony",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        undefined,
        <div
          className="dev-header"
>
          <div
                    className="dev-header-container"
          >
                    <p>
                              Tony
                    </p>
                    <img
                              alt="developer profile picture"
                              className="profile-picture"
                              src={undefined}
                    />
                    <button
                              className="signout-button"
                              onClick={[Function]}
                    >
                              Sign Out
                    </button>
          </div>
</div>,
        <div
          className="main"
>
          <div
                    className="main-container"
          >
                    <p
                              className="main-section-titles"
                    >
                              <NavLink
                                        activeClassName="active"
                                        ariaCurrent="true"
                                        to="/dev-project-list"
                              >
                                        New Project
                              </NavLink>
                               |  
                              <NavLink
                                        activeClassName="active"
                                        ariaCurrent="true"
                                        to="/dev-profile"
                              >
                                        Open Projects
                              </NavLink>
                               |  
                              <NavLink
                                        activeClassName="active"
                                        ariaCurrent="true"
                                        to="/dev-closed-projects"
                              >
                                        Closed Projects
                              </NavLink>
                    </p>
                    <div
                              className="current-projects"
                    >
                              <div
                                        className="info-text"
                              >
                                        <p>
                                                  You currently have no open projects.
                                        </p>
                                        <p>
                                                  Click "New Project" to see avalible projects.
                                        </p>
                              </div>
                    </div>
          </div>
</div>,
      ],
      "className": "profile-background",
    },
    "ref": null,
    "rendered": Array [
      undefined,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="dev-header-container"
>
            <p>
                        Tony
            </p>
            <img
                        alt="developer profile picture"
                        className="profile-picture"
                        src={undefined}
            />
            <button
                        className="signout-button"
                        onClick={[Function]}
            >
                        Sign Out
            </button>
</div>,
          "className": "dev-header",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p>
                Tony
</p>,
              <img
                alt="developer profile picture"
                className="profile-picture"
                src={undefined}
/>,
              <button
                className="signout-button"
                onClick={[Function]}
>
                Sign Out
</button>,
            ],
            "className": "dev-header-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Tony",
              },
              "ref": null,
              "rendered": "Tony",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "developer profile picture",
                "className": "profile-picture",
                "src": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Sign Out",
                "className": "signout-button",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Sign Out",
              "type": "button",
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="main-container"
>
            <p
                        className="main-section-titles"
            >
                        <NavLink
                                    activeClassName="active"
                                    ariaCurrent="true"
                                    to="/dev-project-list"
                        >
                                    New Project
                        </NavLink>
                         |  
                        <NavLink
                                    activeClassName="active"
                                    ariaCurrent="true"
                                    to="/dev-profile"
                        >
                                    Open Projects
                        </NavLink>
                         |  
                        <NavLink
                                    activeClassName="active"
                                    ariaCurrent="true"
                                    to="/dev-closed-projects"
                        >
                                    Closed Projects
                        </NavLink>
            </p>
            <div
                        className="current-projects"
            >
                        <div
                                    className="info-text"
                        >
                                    <p>
                                                You currently have no open projects.
                                    </p>
                                    <p>
                                                Click "New Project" to see avalible projects.
                                    </p>
                        </div>
            </div>
</div>,
          "className": "main",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p
                className="main-section-titles"
>
                <NavLink
                                activeClassName="active"
                                ariaCurrent="true"
                                to="/dev-project-list"
                >
                                New Project
                </NavLink>
                 |  
                <NavLink
                                activeClassName="active"
                                ariaCurrent="true"
                                to="/dev-profile"
                >
                                Open Projects
                </NavLink>
                 |  
                <NavLink
                                activeClassName="active"
                                ariaCurrent="true"
                                to="/dev-closed-projects"
                >
                                Closed Projects
                </NavLink>
</p>,
              <div
                className="current-projects"
>
                <div
                                className="info-text"
                >
                                <p>
                                                You currently have no open projects.
                                </p>
                                <p>
                                                Click "New Project" to see avalible projects.
                                </p>
                </div>
</div>,
            ],
            "className": "main-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    to="/dev-project-list"
>
                    New Project
</NavLink>,
                  " |  ",
                  <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    to="/dev-profile"
>
                    Open Projects
</NavLink>,
                  " |  ",
                  <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    to="/dev-closed-projects"
>
                    Closed Projects
</NavLink>,
                ],
                "className": "main-section-titles",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "activeClassName": "active",
                    "ariaCurrent": "true",
                    "children": "New Project",
                    "to": "/dev-project-list",
                  },
                  "ref": null,
                  "rendered": "New Project",
                  "type": [Function],
                },
                " |  ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "activeClassName": "active",
                    "ariaCurrent": "true",
                    "children": "Open Projects",
                    "to": "/dev-profile",
                  },
                  "ref": null,
                  "rendered": "Open Projects",
                  "type": [Function],
                },
                " |  ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "activeClassName": "active",
                    "ariaCurrent": "true",
                    "children": "Closed Projects",
                    "to": "/dev-closed-projects",
                  },
                  "ref": null,
                  "rendered": "Closed Projects",
                  "type": [Function],
                },
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <div
                  className="info-text"
>
                  <p>
                                    You currently have no open projects.
                  </p>
                  <p>
                                    Click "New Project" to see avalible projects.
                  </p>
</div>,
                "className": "current-projects",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <p>
                      You currently have no open projects.
</p>,
                    <p>
                      Click "New Project" to see avalible projects.
</p>,
                  ],
                  "className": "info-text",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "You currently have no open projects.",
                    },
                    "ref": null,
                    "rendered": "You currently have no open projects.",
                    "type": "p",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Click \\"New Project\\" to see avalible projects.",
                    },
                    "ref": null,
                    "rendered": "Click \\"New Project\\" to see avalible projects.",
                    "type": "p",
                  },
                ],
                "type": "div",
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          undefined,
          <div
            className="dev-header"
>
            <div
                        className="dev-header-container"
            >
                        <p>
                                    Tony
                        </p>
                        <img
                                    alt="developer profile picture"
                                    className="profile-picture"
                                    src={undefined}
                        />
                        <button
                                    className="signout-button"
                                    onClick={[Function]}
                        >
                                    Sign Out
                        </button>
            </div>
</div>,
          <div
            className="main"
>
            <div
                        className="main-container"
            >
                        <p
                                    className="main-section-titles"
                        >
                                    <NavLink
                                                activeClassName="active"
                                                ariaCurrent="true"
                                                to="/dev-project-list"
                                    >
                                                New Project
                                    </NavLink>
                                     |  
                                    <NavLink
                                                activeClassName="active"
                                                ariaCurrent="true"
                                                to="/dev-profile"
                                    >
                                                Open Projects
                                    </NavLink>
                                     |  
                                    <NavLink
                                                activeClassName="active"
                                                ariaCurrent="true"
                                                to="/dev-closed-projects"
                                    >
                                                Closed Projects
                                    </NavLink>
                        </p>
                        <div
                                    className="current-projects"
                        >
                                    <div
                                                className="info-text"
                                    >
                                                <p>
                                                            You currently have no open projects.
                                                </p>
                                                <p>
                                                            Click "New Project" to see avalible projects.
                                                </p>
                                    </div>
                        </div>
            </div>
</div>,
        ],
        "className": "profile-background",
      },
      "ref": null,
      "rendered": Array [
        undefined,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="dev-header-container"
>
              <p>
                            Tony
              </p>
              <img
                            alt="developer profile picture"
                            className="profile-picture"
                            src={undefined}
              />
              <button
                            className="signout-button"
                            onClick={[Function]}
              >
                            Sign Out
              </button>
</div>,
            "className": "dev-header",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <p>
                  Tony
</p>,
                <img
                  alt="developer profile picture"
                  className="profile-picture"
                  src={undefined}
/>,
                <button
                  className="signout-button"
                  onClick={[Function]}
>
                  Sign Out
</button>,
              ],
              "className": "dev-header-container",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Tony",
                },
                "ref": null,
                "rendered": "Tony",
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "developer profile picture",
                  "className": "profile-picture",
                  "src": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Sign Out",
                  "className": "signout-button",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "Sign Out",
                "type": "button",
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="main-container"
>
              <p
                            className="main-section-titles"
              >
                            <NavLink
                                          activeClassName="active"
                                          ariaCurrent="true"
                                          to="/dev-project-list"
                            >
                                          New Project
                            </NavLink>
                             |  
                            <NavLink
                                          activeClassName="active"
                                          ariaCurrent="true"
                                          to="/dev-profile"
                            >
                                          Open Projects
                            </NavLink>
                             |  
                            <NavLink
                                          activeClassName="active"
                                          ariaCurrent="true"
                                          to="/dev-closed-projects"
                            >
                                          Closed Projects
                            </NavLink>
              </p>
              <div
                            className="current-projects"
              >
                            <div
                                          className="info-text"
                            >
                                          <p>
                                                        You currently have no open projects.
                                          </p>
                                          <p>
                                                        Click "New Project" to see avalible projects.
                                          </p>
                            </div>
              </div>
</div>,
            "className": "main",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <p
                  className="main-section-titles"
>
                  <NavLink
                                    activeClassName="active"
                                    ariaCurrent="true"
                                    to="/dev-project-list"
                  >
                                    New Project
                  </NavLink>
                   |  
                  <NavLink
                                    activeClassName="active"
                                    ariaCurrent="true"
                                    to="/dev-profile"
                  >
                                    Open Projects
                  </NavLink>
                   |  
                  <NavLink
                                    activeClassName="active"
                                    ariaCurrent="true"
                                    to="/dev-closed-projects"
                  >
                                    Closed Projects
                  </NavLink>
</p>,
                <div
                  className="current-projects"
>
                  <div
                                    className="info-text"
                  >
                                    <p>
                                                      You currently have no open projects.
                                    </p>
                                    <p>
                                                      Click "New Project" to see avalible projects.
                                    </p>
                  </div>
</div>,
              ],
              "className": "main-container",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <NavLink
                      activeClassName="active"
                      ariaCurrent="true"
                      to="/dev-project-list"
>
                      New Project
</NavLink>,
                    " |  ",
                    <NavLink
                      activeClassName="active"
                      ariaCurrent="true"
                      to="/dev-profile"
>
                      Open Projects
</NavLink>,
                    " |  ",
                    <NavLink
                      activeClassName="active"
                      ariaCurrent="true"
                      to="/dev-closed-projects"
>
                      Closed Projects
</NavLink>,
                  ],
                  "className": "main-section-titles",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "activeClassName": "active",
                      "ariaCurrent": "true",
                      "children": "New Project",
                      "to": "/dev-project-list",
                    },
                    "ref": null,
                    "rendered": "New Project",
                    "type": [Function],
                  },
                  " |  ",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "activeClassName": "active",
                      "ariaCurrent": "true",
                      "children": "Open Projects",
                      "to": "/dev-profile",
                    },
                    "ref": null,
                    "rendered": "Open Projects",
                    "type": [Function],
                  },
                  " |  ",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "activeClassName": "active",
                      "ariaCurrent": "true",
                      "children": "Closed Projects",
                      "to": "/dev-closed-projects",
                    },
                    "ref": null,
                    "rendered": "Closed Projects",
                    "type": [Function],
                  },
                ],
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <div
                    className="info-text"
>
                    <p>
                                        You currently have no open projects.
                    </p>
                    <p>
                                        Click "New Project" to see avalible projects.
                    </p>
</div>,
                  "className": "current-projects",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <p>
                        You currently have no open projects.
</p>,
                      <p>
                        Click "New Project" to see avalible projects.
</p>,
                    ],
                    "className": "info-text",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "You currently have no open projects.",
                      },
                      "ref": null,
                      "rendered": "You currently have no open projects.",
                      "type": "p",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Click \\"New Project\\" to see avalible projects.",
                      },
                      "ref": null,
                      "rendered": "Click \\"New Project\\" to see avalible projects.",
                      "type": "p",
                    },
                  ],
                  "type": "div",
                },
                "type": "div",
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
